### ISSUE TITLE
Mac Silicon support

### ISSUE THREAD

**Issue by kiyoon** [Electrostatics/apbs#289]
Is there a chance to make a release with macOS arm64 binary? I can't seem to make it work. Thank you!

**Comment by sobolevnrm** [comment]
If the architecture you're looking for isn't on https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories, then I'm not sure we can support it.  I don't have access to a Mac OS ARM64 system.

**Comment by kiyoon** [comment]
@sobolevnrm The M1 processor is what I want and is available!

**Comment by kiyoon** [comment]
Hi, when can I expect to see the support? Thank you.

**Comment by jbardhan** [comment]
Hi,

Thank you for the correction -- that certainly makes it an easier task.  I'm replying so you knew we saw your updated request, but I am not sure when exactly we'll release a version supporting M1. @sobolevnrm may have more to say. 

Best regards,
Jay

**Comment by sobolevnrm** [comment]
> Hi, when can I expect to see the support? Thank you.

The short answer may be "never" if we can't get additional support (financial or volunteer) for this project.  I'm sorry--that is probably not the answer you were looking for.  I will continue to try to find the time for this but am not sure when I'll get to it.

**Comment by YoshitakaMo** [comment]
@kiyoon I've created Homebrew formula for APBS 3.4.1. Now you can install `apbs` binary on Apple silicon with `brew install brewsci/bio/apbs`:beer:

**Comment by kiyoon** [comment]
@YoshitakaMo Thank you, this is amazing!!

In APBS release in this repo, there's `share/apbs/tools/bin` directory with a few useful cli tools included as well. In my case I use `multivalue`. Is there a chance this can also be included?

**Comment by YoshitakaMo** [comment]
@kiyoon 

> Is there a chance this can also be included?

Yes!! The tools were not built during the last installation (`3.4.1_1`) because I've turned it off `-DBUILD_TOOLS=OFF`.
Now I've regenerated the Homebrew bottle with `-DBUILD_TOOLS=ON`. `apbs` version 3.4.1_2 is now available. Try `brew update ; brew upgrade brewsci/bio/apbs`
You may add `PATH` to use them (because the names of the binaries were so common that I was afraid of conflicting with other Homebrew binaries.)

```
==> Caveats
==> apbs
Some useful tools have been also installed in /opt/homebrew/opt/apbs/share/apbs/tools.
You may want to add the following to your .bashrc or .zshrc to use these tools:
  export APBS_TOOLS_HOME=$(brew --prefix)/opt/apbs/share/apbs/tools/
  export PATH=${PATH}:${APBS_TOOLS_HOME}/bin
==> Summary
🍺  /opt/homebrew/Cellar/apbs/3.4.1_2: 662 files, 88.3MB, built in 17 seconds
```

**Comment by kiyoon** [comment]
@YoshitakaMo This is perfect! I confirmed it's running on my machine with no problem. Thanks a lot for your contribution!!

@sobolevnrm You can close the issue now if you don't plan to add support. However, I think you might want to add the official binaries in the future. Either way, I'm happy now. :)

**Comment by kiyoon** [comment]
Thanks to @YoshitakaMo, it is so much easier to install and use APBS now!

However, building with brew takes time and it also installs so many build-time dependencies. Also I don't use brew on Linux, and I wanted a static binary release that just simply works, regardless of the platform.

So I've repackaged it as a python package, including the binary corresponding to the platform you use.

```bash
pip install apbs-binary==3.4.1.1
```

```python
from apbs_binary import run_apbs, popen_apbs, run_multivalue, popen_multivalue

run_apbs("--help")  # like subprocess.run(["apbs", "-h"])
popen_apbs("--help")  # like subprocess.Popen(["apbs", "-h"])

# For no-argument case, pass an empty string
run_apbs("")  # like subprocess.run(["apbs"])

# Pass a list of arguments
run_apbs(["--output-format=xml", "input.in"])

# Run multivalue
run_multivalue(...)
popen_multivalue(...)
```

Currently it supports Linux and Mac (intel and Apple Silicon), but not Windows. I couldn't figure out the dependency problem with `python39.dll` and if anyone can help me, I'd like to support Windows too.

See https://github.com/deargen/py-apbs-binary for more.
